engine/src/main/java/org/pentaho/di/repository/filerep/KettleFileRepository.java

```
@@ -0,0 +1,1571 @@
+
+    return id.getId().substring( slashIndex + 1, dotIndex );
+  }
+
+  @Override
+  public ObjectId renameJob( ObjectId id_job, RepositoryDirectoryInterface newDir, String newName )
+    throws KettleException {
+    return renameJob( id_job, null, newDir, newName );
+  }
+
+  @Override
+  public ObjectId renameJob( ObjectId id_job, String versionComment, RepositoryDirectoryInterface newDir,
+    String newName ) throws KettleException {
+    ObjectId objectId = renameObject( id_job, newDir, newName, EXT_JOB );
+    if ( !Utils.isEmpty( versionComment ) ) {
+      insertLogEntry( "Rename job : " + versionComment );
+    }
+    return objectId;
+  }
+
+  @Override
+  public ObjectId renameRepositoryDirectory( ObjectId id, RepositoryDirectoryInterface newParentDir, String newName ) throws KettleException {
+    if ( newParentDir != null || newName != null ) {
+      try {
+        // In case of a root object, the ID is the same as the relative filename...
+        RepositoryDirectoryInterface tree = loadRepositoryDirectoryTree();
+        RepositoryDirectoryInterface dir = tree.findDirectory( id );
+
+        if ( dir == null ) {
+          throw new KettleException( "Could not find folder [" + id + "]" );
+        }
+
+        // If newName is null, keep the current name
+        newName = ( newName != null ) ? newName : dir.getName();
+
+        FileObject folder = KettleVFS.getFileObject( dir.getPath() );
+
+        String newFolderName = null;
+
+        if ( newParentDir != null ) {
+          FileObject newParentFolder = KettleVFS.getFileObject( newParentDir.getPath() );
+
+          newFolderName = newParentFolder.toString() + "/" + newName;
+        } else {
+          newFolderName = folder.getParent().toString() + "/" + newName;
+        }
+
+        FileObject newFolder = KettleVFS.getFileObject( newFolderName );
+        folder.moveTo( newFolder );
+
+        return new StringObjectId( dir.getObjectId() );
+      } catch ( Exception e ) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)